src/utils/getApplicationServerSideProps.js

```diff
@@ -21,9 +21,12 @@ const getApplicationServerSideProps = (callbackFn) => async ({
     cookies.remove('user');
     cookies.remove('token');
 
+    req.headers.referer =
+      req.headers.referer ?? `http://${req.headers.host}${req.url}`;
```

Comment:
Shouldn't this be https:// ?

```suggestion
      req.headers.referer ?? `https://${req.headers.host}${req.url}`;
```

Reply:
Https won't work locally.

I wonder if this is correct: https://github.com/vercel/next.js/issues/2469#issuecomment-313194091
@bertramakers @willaerk

Reply:
I would allow both methods if possible:
* If "locally" means you just building and serving the frontend and accessing it through http://localhost:3000, then the referrer will contain http://
* If "locally" means building the frontend and serving it from your vagrant box, the referrer will contain https://

Reply:
He is the creator of Next.js so I'm pretty sure he knows some things 😄

Reply:
Would checking `process.env.NODE_ENV === 'production'` be enough here?

When it's a production build use `https` else `http`

Reply:
You could use the X-Forwarded-Proto header to get the protocol I think.

Reply:
Locally I don't get that header:

```
headers {
  host: 'localhost:3000',
  'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:89.0) Gecko/20100101 Firefox/89.0',
  accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
  'accept-language': 'en-US,nl;q=0.7,en;q=0.3',
  'accept-encoding': 'gzip, deflate',
  connection: 'keep-alive',
  'upgrade-insecure-requests': '1',
  'cache-control': 'max-age=0'
}
```

I'll check if i'ts there with production build

Reply:
If it is not there, it means there was no forwarding, and we can assume http:// was used I guess.

Reply:
> When it's a production build use https else http

I remember from the permanent redirects discussion that you and @geoffreydhuyvetters  sometimes also run production mode on localhost so production mode does not necessarily mean https.

Aside from that, is the referer even relevant to this discussion? Isn't Bram's question about the URL to use when there's _no_ referer? So the default URL to redirect to after login?

The way I see it, we always redirect http:// requests to https:// anyway on the box & servers so http:// would work for both localhost and the Vagrant box and servers?

Reply:
For example browse to http://www.uitdatabank.be/ and you automatically land on the https version.

Reply:
> Locally I don't get that header

Do you mean "locally" on localhost or "locally" in the Vagrant box?

I'm pretty sure it will only be included when you run it in the Vagrant box. The nginx server in the box will set that header.

Reply:
I meant on localhost by `locally`

Reply:
Either way on the production build something is broken, so I'll have to fix it first before we merge

Reply:
That would probably explain why the header is not there then, since there's no forwarding happening by a reverse proxy that handles nginx. If the header is not there I think you can safely assume the protocol is http like mentioned in the link you shared. If the header is there you can use the protocol specified in it.

But I'm still not sure if it even matters since we redirect everything from http:// to https:// anyway. @willaerk am I overlooking something in that assumption?

Reply:
I've implemented a check for the **X-Forwarded-Proto** header:

https://github.com/cultuurnet/udb3-frontend/pull/247/commits/1cdefbb428aa828b89ec061eaaa74a562e68f585

Reply:
We do redirect http:// to https:// for all requests, but maybe it is not the most "elegant" solution to always rely on that being available. After all, the X-Forwarded-Proto header is used specifically for determining the protocol of the original request.

Reply:
Alright is this logic correct then?

- If there's no x-forwarded-proto header the request was not forwarded by e.g. nginx and the app is _probably_* running on localhost so we can use `http://` in the link
- If the header is there, we can use the protocol defined in it

*But what if nginx stops sending this header for some reason?

This is also how @simon-debruijn has implemented it now (and also what I suggested above)

Reply:
If the header isn't sent by nginx the redirect from http to https will still catch this no?

Reply:
Yes definitely, I just think that the `X-Forwarded-Proto` in itself is also not the most elegant solution. 😄  But at least if one of the two measures breaks we can still rely on the other one.